src/commonMain/kotlin/org/kbson/BsonDecimal128.kt

```diff
@@ -0,0 +1,88 @@
+/*
+ * Copyright 2008-present MongoDB, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.kbson
+
+/** A representation of the BSON Decimal128 type. */
+class BsonDecimal128(val high: Long, val low: Long) : BsonValue() {
```

Comment:
Is it worth trying to ease integration with BigDecimal (or equivalent) in environments where it's available?

In particular, in the Java implementation there is a public method `Decimal128#bigDecimalValue` which obviously can't be included here, but it is built on top of two private methods that potentially _could_ be made public (`Decimal128#getExponent` and `Decimal128#getBytes`) in this implementation.

Reply:
Happy to take a lead from the Realm team - we can always look to improve integration in the future as well.

Reply:
I don't think we have any strong opinions on this. I see this class as something that is much less used than a full `Decimal128` class, like the one found in the Java library.

Maybe I don't fully understand what this type does exactly that `Decimal128` doesn't. But I read it as a backing type , closer to the actual bytes, whereas `Decimal128` more contains a usable public API.

If that is the case, just having this constructor and exposing whatever we need on `Decimal128` seems like the right approach.

Reply:
I think the plan was not to have a Decimal128 class at all, but rather to embed any of its functionality directly in BsonDecimal128, the same as is currently done with BsonObjectId.  I'm not sure whether that's the right direction.  If we end up adding support for Kotlin data classes, I think we would want the field types not to be BsonDecimal128 or BsonObjectId, but rather Decimal128 and ObjectId as in a typical Java POJO.

What do you think, Ross?

Reply:
Hmm, that would be pretty annoying on our end as all other SDKs currently have types named `ObjectId` and `Decimal128`, including Realm Java. So having to use `BsonDecimal128` in Kotlin would be confusing IMO.

Reply:
I'm wondering if we can alias the type (not sure if that works for KMP) or just mirror the functionality.

Currently porting just BsonValues so I had planned to make the BsonObjectId and BsonDecimal128 functionally equivalent to the existing ObjectId and Decimal128 types.

If we do end up adding separate ObjectId and Decimal128 classes - they can just be backed by the Bson types easily enough.

Reply:
We just talked about this on the Kotlin team, and maybe just having a type alias might be good enough, but that does require that the `BsonObjectId` and `BsonDecimal128` API are functionally equivalent to the current Java `ObjectId` and `Decimal128`. I haven't looked that deep into their differences, so that might already be the case.

Reply:
Sounds fine with me 👍